internal/featureflags/featureflags.go

```diff
@@ -0,0 +1,75 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package featureflags
+
+import "os"
+
+// FeatureFlag represents a command line switch to turn certain features
+// ON or OFF. Values are read from environment variables defined by
+// the feature flag. The feature flag can have default values that are used
+// when the resp. environment variable does not exist
+// Feature flags are considered to be "enabled" if their resp. environment variable
+// is set (to some value, usually "1").
+// Feature flags are considered to be "disabled" if their resp. environment variable
+// is set to "0" explicitly.
+type FeatureFlag struct {
```

Comment:
Should we maybe check for better values here?
1, true for enabled
0, false for disabled

```
// Feature flags are considered to be "disabled" if their resp. environment variable
// is set to "0" explicitly.
```

so setting MONAOC_FF=false would enable the ff currently

Reply:
open to discussing that.
for me it is fine, just continued with the semantics that where already there.

Reply:
Question is if we should panic whether the value is completely invalid. Imo would be ok.

Reply:
> https://pkg.go.dev/strconv#ParseBool
> I think we should use it, as it contains the current use-case, but also good other ones

looks good, let's do it like that :)